flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/declarative/ExecutingTest.java

```diff
@@ -0,0 +1,444 @@
+            finishedStateValidator.activate(asserter);
+        }
+
+        public void setHowToHandleFailure(Function<Throwable, Executing.FailureResult> function) {
+            this.howToHandleFailure = function;
+        }
+
+        public void setCanScaleUp(Supplier<Boolean> supplier) {
+            this.canScaleUp = supplier;
+        }
+
+        public void setGetMainThreadExecutor(Supplier<Executor> supplier) {
+            this.getMainThreadExecutor = supplier;
+        }
+
+        public void setExpectedStateChecker(Function<State, Boolean> function) {
+            this.expectedStateChecker = function;
+        }
+
+        // --------- Interface Implementations ------- //
+
+        @Override
+        public void goToCanceling(
+                ExecutionGraph executionGraph,
+                ExecutionGraphHandler executionGraphHandler,
+                OperatorCoordinatorHandler operatorCoordinatorHandler) {
+            cancellingStateValidator.validateInput(
+                    new CancellingArguments(
+                            executionGraph, executionGraphHandler, operatorCoordinatorHandler));
+        }
+
+        @Override
+        public Executing.FailureResult howToHandleFailure(Throwable failure) {
+            return howToHandleFailure.apply(failure);
+        }
+
+        @Override
+        public boolean canScaleUp(ExecutionGraph executionGraph) {
+            return canScaleUp.get();
+        }
+
+        @Override
+        public void goToRestarting(
+                ExecutionGraph executionGraph,
+                ExecutionGraphHandler executionGraphHandler,
+                OperatorCoordinatorHandler operatorCoordinatorHandler,
+                Duration backoffTime) {
+            restartingStateValidator.validateInput(
+                    new RestartingArguments(
+                            executionGraph,
+                            executionGraphHandler,
+                            operatorCoordinatorHandler,
+                            backoffTime));
+        }
+
+        @Override
+        public void goToFailing(
+                ExecutionGraph executionGraph,
+                ExecutionGraphHandler executionGraphHandler,
+                OperatorCoordinatorHandler operatorCoordinatorHandler,
+                Throwable failureCause) {
+            failingStateValidator.validateInput(
+                    new FailingArguments(
+                            executionGraph,
+                            executionGraphHandler,
+                            operatorCoordinatorHandler,
+                            failureCause));
+        }
+
+        @Override
+        public void runIfState(State expectedState, Runnable action) {
+            if (expectedStateChecker.apply(expectedState)) {
+                action.run();
+            }
+        }
+
+        @Override
+        public boolean isState(State expectedState) {
+            return false;
```

Comment:
This should until `true` until we transitioned to another state

Reply:
This method is actually not covered by the tests (since it's used when triggeringSavepoints, which I suppose should be tested in an declarative scheduler IT case)